medicines/api/src/document.rs

```diff
@@ -10,34 +15,47 @@ pub struct Document {
     title: Option<String>,
     highlights: Option<Vec<String>>,
     created: Option<String>,
-    doc_type: Option<String>, // TODO: use DocumentType enum below
+    doc_type: Option<DocumentType>,
     file_size_in_bytes: Option<i32>,
     name: Option<String>,
     url: Option<String>,
 }
 
 impl Document {
-    pub fn is_doc_type(&self, doc_type: &DocumentType) -> bool {
-        self.doc_type == Some(doc_type.to_search_str().to_owned())
+    pub fn is_doc_type(&self, doc_type: DocumentType) -> bool {
+        self.doc_type == Some(doc_type)
+    }
+
+    pub fn product_name(&self) -> Option<&str> {
+        self.product_name.as_deref()
+    }
+
+    pub fn substances(&self) -> impl Iterator<Item = &str> {
+        self.active_substances
+            .iter()
+            .flat_map(|s| s.iter())
+            .map(|s| s.as_str())
     }
 }
 
-impl From<IndexResult> for Document {
-    fn from(r: IndexResult) -> Self {
-        Self {
+impl TryFrom<IndexResult> for Document {
+    type Error = DocTypeParseError;
+
+    fn try_from(r: IndexResult) -> Result<Self, Self::Error> {
+        Ok(Self {
             product_name: r.product_name,
             active_substances: Some(r.substance_name),
             title: Some(r.title),
             created: r.created,
-            doc_type: Some(r.doc_type),
+            doc_type: Some(r.doc_type.parse()?),
```

Comment:
Given that `doc_type` is `Option`, I believe [we can use `.ok()`][ok] here and we'd be able to implement `From` instead of `TryFrom`:

```suggestion
            doc_type: r.doc_type.ok(),
```

[`result.ok()`][ok] is like a reverse of [`option.ok_or(error)`][ok or], both of them seem really useful!

[ok]: https://doc.rust-lang.org/std/result/enum.Result.html#method.ok
[ok or]: https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or